Kernel/Devices/KCOVInstance.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021, Patrick Meyer <git@the-space.agency>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/SpinLock.h>
+#include <Kernel/VM/AnonymousVMObject.h>
+#include <sys/kcov.h>
+
+namespace Kernel {
+
+/*
+ * One KCOVInstance is allocated per process, when the process opens /dev/kcov
+ * for the first time. At this point it is in state OPENED. When a thread in
+ * the same process then uses the KCOV_ENABLE ioctl on the block device, the
+ * instance enters state TRACING.
+ * 
+ * A KCOVInstance in state TRACING can return to state OPENED by either the
+ * KCOV_DISABLE ioctl or by killing the thread. A KCOVInstance in state OPENED
+ * can return to state UNUSED only when the process dies. At this point nothing
+ * reference the KCOVInstance any longer and it will be deleted. 
+ */
+class KCOVInstance final {
+public:
+    KCOVInstance(ProcessID pid);
+
+    int buffer_allocate(size_t buffer_size_in_entries);
+    void buffer_add_pc(u64 pc);
+
+    kcov_pc_t* buffer;
+    SpinLock<u8> lock;
+    enum {
+        UNUSED = 0,
+        OPENED = 1,
+        TRACING = 2,
+    } state;
+
+    AK::RefPtr<Kernel::AnonymousVMObject> vmobject;
+
+private:
+    ProcessID pid;
+    u64 buffer_size_in_entries;
+    size_t buffer_size_in_bytes;
```

Comment:
Private instance fields should be prefixed with `m_`. Also, primitive types should be initialized.

Reply:
like this?

Reply:
I see now, thanks.